src/app/js/formats/streamgraph/utils.js

```diff
@@ -0,0 +1,159 @@
+import differenceBy from "lodash/differenceBy";
+import * as d3 from "d3";
+
+export function zoomFunction() {
+  // create new scale ojects based on event
+  let new_xScale = d3.event.transform.rescaleX(this.xAxisScale);
+  let new_yScale = d3.event.transform.rescaleY(this.yAxisScale);
+
+  // update axes
+  this.gx.call(this.xAxis.scale(new_xScale));
+  this.gyr.call(this.yAxisR.scale(new_yScale));
+  this.gyl.call(this.yAxisL.scale(new_yScale));
+
+  // update streams
+  this.streams.attr("transform", d3.event.transform);
+}
+
+export function distinctColors(count) {
+  let colors = [];
+  for (let hue = 0; hue < 360; hue += 360 / count) {
+    colors.push(hslToHex(hue, 90, 50));
+  }
+  return colors;
+}
+
+export function hslToHex(h, s, l) {
+  h /= 360;
+  s /= 100;
+  l /= 100;
+  let r, g, b;
+  if (s === 0) {
+    r = g = b = l; // achromatic
+  } else {
+    const hue2rgb = (p, q, t) => {
+      if (t < 0) t += 1;
+      if (t > 1) t -= 1;
+      if (t < 1 / 6) return p + (q - p) * 6 * t;
+      if (t < 1 / 2) return q;
+      if (t < 2 / 3) return p + (q - p) * (2 / 3 - t) * 6;
+      return p;
+    };
+    const q = l < 0.5 ? l * (1 + s) : l + s - l * s;
+    const p = 2 * l - q;
+    r = hue2rgb(p, q, h + 1 / 3);
+    g = hue2rgb(p, q, h);
+    b = hue2rgb(p, q, h - 1 / 3);
+  }
+  const toHex = x => {
+    const hex = Math.round(x * 255).toString(16);
+    return hex.length === 1 ? "0" + hex : hex;
+  };
+  return `#${toHex(r)}${toHex(g)}${toHex(b)}`;
+}
+
+export function transformDataIntoMapArray(formatData) {
+  let dateMin = -42;
+  let dateMax = -42;
+  let valuesObjectsArray = [];
+
+  if (formatData) {
+    for (let i = 0; i < formatData.length; i++) {
+      let elem = formatData[i];
+      if (elem) {
+        // source: "2010"
+        // target: "Pétrogenèse de roches basaltiques"
+        // weight: 0.9013102637325718
+
+        if (elem.source && elem.weight) {
+          let currentItem = undefined;
+
+          for (let tmpElem of valuesObjectsArray) {
+            if (tmpElem.name === elem.target) {
+              currentItem = tmpElem;
+            }
+          }
+
+          if (currentItem === undefined) {
+            currentItem = {
+              name: elem.target,
+              values: []
+            };
+            valuesObjectsArray.push(currentItem);
+          }
+
+          currentItem.values.push({
+            date: new Date(elem.source),
+            value: elem.weight
+          });
+
+          if (parseInt(elem.source) < dateMin || dateMin === -42) {
```

Comment:
parseInt is unsafe and needs to have a second argument to set the base. `parseInt(elem.source, 10)`